tutorials/dataframe/distrdf001_spark_connection.py

```diff
@@ -0,0 +1,88 @@
+## setup, an RDataFrame with distributed capabilities is created and connected
+## to the SparkContext instance. Finally, a couple of histograms are drawn from
+## the created columns in the dataset.
+##
+## \macro_code
+## \macro_image
+##
+## \date March 2021
+## \author Vincenzo Eduardo Padulano
+import os
+import sys
+import pyspark
+import ROOT
+
+# Point RDataFrame calls to Spark RDataFrame object
+RDataFrame = ROOT.RDF.Experimental.Distributed.Spark.RDataFrame
+
+# Setup the connection to Spark
+# First create a dictionary with keys representing Spark specific configuration
+# parameters. In this tutorial we use the following configuration parameters:
+#
+#     1. spark.app.name: The name of the Spark application
+#     2. spark.master: The Spark endpoint responsible for running the
+#         application. With the syntax "local[4]" we signal Spark we want to run
+#         locally on the same machine with 4 cores, each running a separate
+#         process. The default behaviour of a Spark application would run
+#         locally on the same machine with as many concurrent processes as
+#         available cores, that could be also written as "local[*]".
+#     3. PYSPARK_PYTHON: Environment variable pointing to the path to the Python
+#         executable that will run both in the Spark driver (i.e. the process
+#         running this script) and the Spark executors (i.e. the processes that
+#         will run the computations). In order for Spark to work correctly, the
+#         environment of both types of processes should have the same Python
+#         version and satisfy the same dependencies, so we choose to point this
+#         variable to the same Python executable that is running this script.
+#
+# If you have access to a remote cluster you should substitute the endpoint URL
+# of your spark master in the form "spark://HOST:PORT" in the value of
+# `spark.master`. Depending on the availability of your cluster you may request
+# more computing nodes or cores per node with a similar configuration:
+#
+#     confdict = {"spark.master": "spark://HOST:PORT",
+#                 "spark.executor.instances": <number_of_nodes>,
+#                 "spark.executor.cores" <cores_per_node>,}
+#
+# Depending on the cluster configuration, you may need to adjust more settings,
+# for example `spark.pyspark.python` as it is done in this tutorial. You can
+# find all configuration options and more details in the Spark documentation at
+# https://spark.apache.org/docs/latest/configuration.html .
+confdict = {"spark.app.name": "distrdf001_spark_connection",
+            "spark.master": "local[4]",}
+os.environ["PYSPARK_PYTHON"] = sys.executable
+# The configuration dictionary can be digested by a SparkConf object
+sparkconf = pyspark.SparkConf().setAll(confdict.items())
+# Finally create a SparkContext with the configuration stored in `sparkconf`
+sc = pyspark.SparkContext(conf=sparkconf)
```

Comment:
The Spark specifics look a little bit scary here. Why not compressing it with some nice Python syntax? I see that you want to explain in detail how to setup the Spark side, but probably it would be worth to put things together:

```python
sparkconf = pyspark.SparkConf().setAll({
         "spark.app.name": "distrdf001_spark_connection",
         "spark.master": "local[4]"
        }.items())
sc = pyspark.SparkContext(conf=sparkconf)
```

Reply:
And probably I would name `sc` just `sparkcontext`. Spelling this out makes it very clear what it is, without remembering the name of the class used for the ctor.